pkg/providers/instance/instance.go

```diff
@@ -78,15 +83,18 @@ func NewProvider(ctx context.Context, region string, ec2api ec2iface.EC2API, una
 		subnetProvider:         subnetProvider,
 		launchTemplateProvider: launchTemplateProvider,
 		ec2Batcher:             batcher.EC2(ctx, ec2api),
+		recorder:               recorder,
 	}
 }
 
 func (p *Provider) Create(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType) (*Instance, error) {
 	schedulingRequirements := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...)
+	resolvedNodeClassITs := instanceTypes
```

Comment:
I'm not 100% sure, but since this is a slice, I think this just stores the pointer to the front of instanceTypes, so doing something like this doesn't actually copy the slice like I think you intend

Reply:
Further, I'm not sure if you need to copy the slice here. How were you thinking this would help the eventing?

Reply:
Hmmm, well the reason I tried to store that is because we are not publishing an event until we filter the exotic types. But we update the instance types [here](https://github.com/aws/karpenter-provider-aws/blob/main/pkg/providers/instance/instance.go#L88) and if I don't store it before that then we will not get the previously filtered instance types and like I mentioned earlier, the ITs passed to this function have also gone through one round of [filtering](https://github.com/aws/karpenter-provider-aws/blob/main/pkg/cloudprovider/cloudprovider.go#L247).

Reply:
Right, but I'm not sure that you actually need the previously filtered instance types, I think you just need the result at the end and you also need to capture the number of instance types that are filtered at each round